### PR TITLE
test for large request body

### PR DESCRIPTION
**Pull request by TeodoroFilippini**
@mristin 

LMK what you think about this test. It works, but there is no check that it does; should I add some sort of timeout in the pursuit of proving that the relay server is NOT attempting to parse the message?

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -68,7 +68,9 @@ The following external dependencies are used in the Go code:
 The following tools are used for code generation:
 - [swagger-to]("https://github.com/Parquery/swagger-to") for converting Swagger files to Go server and Py client code;
 - [icontract]("https://github.com/Parquery/icontract") for design-by-contract in Python.
-- [protoc-gen-go]("https://github.com/golang/protobuf/tree/master/protoc-gen-go") for convertig .proto files to Go code.
+- [protoc-gen-go]("https://github.com/golang/protobuf/tree/master/protoc-gen-go") for converting .proto files to Go 
+code.
```
Comment (mristin) [inline CONTRIBUTING.md]: I think you need to indent here?

File: tests/request_size_test.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    proc = subprocess.Popen(cmd, stdout=stdout)
+    with tests.proc.terminating(proc=proc, timeout=5):
+        # let the server initialize
+        tests.proc.sleep_while_process(proc=proc, seconds=1)
+
+        if proc.poll() is not None:
+            raise AssertionError("Expected the server process to be alive, but it died.")
+
+        # relay a message with wrong auth token and large attachment (size: 1 Gb)
+        with tempfile.NamedTemporaryFile() as fid:
```
Comment (mristin) [inline tests/request_size_test.py]: You actually don't have to create the file. You only need to figure out which functions `MultipartEncoder` will actually use.

Then you can just create a mock class with these methods that returns a couple of zero bytes and sleeps for ten seconds before sending the other chunk of a couple of zero bytes.

You can then test that the time elapse for `post()` was less than, say, 2 seconds.

**Comment by TeodoroFilippini** [comment]
@mristin 

Here's the test with sockets sending raw bytes. The socket is *not* automatically closed by the server upon failed authentication; the connection is by default as `Connection: close` needs to be in all HTTP request headers as per HTTP 1.1: according to https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html,

> HTTP/1.1 applications that do not support persistent connections MUST include the "close" connection option in every message. 

As far as I checked, we do not support persistent connection on our server. I tried sending more data after the failed authentication and it does go through, although the server doesn't seem to receive it.

**Comment by mristin** [comment]
Hi Teodoro,
Please have a look at this stackoverflow:
https://stackoverflow.com/questions/4899593/python-doesnt-detect-a-closed-socket-until-the-second-send

I think that in your case it means that you send a short file and check if
you get the acknowledgement. Please double-check with HTTP specs.

Cheers Marko

**Comment by mristin** [comment]
P.S. see also
https://stackoverflow.com/questions/48613006/python-sendall-not-raising-connection-closed-error

**Comment by mristin** [comment]
Thanks, LGTM! (see the remark above before merging)

**Pull request merged**